GH-3897: Deprecate `ChannelSecurityInterceptor` (#3915)
spring-projects/spring-integration: src/reference/asciidoc/security.adoc

We can either trust all those messages or secure our service against "`infecting`" messages.

Spring Integration, together with https://projects.spring.io/spring-security/[Spring Security], provides a simple and comprehensive way to secure message channels, as well as other part of the integration solution.

> Starting with version 6.0, the `ChannelSecurityInterceptor` as well as its configuration via `@SecuredChannel` annotation and  XML `<secured-channels>` have been deprecation in favor of `AuthorizationChannelInterceptor` from `spring-security-messaging` module.

Reviewer: ```suggestion
Starting with version 6.0, the `ChannelSecurityInterceptor` as well as its configuration via `@SecuredChannel` annotation and  XML `<secured-channels>` have been deprecation in favor of using `AuthorizationChannelInterceptor` from the `spring-security-messaging` module.
```

> The respective `AuthorizationManager` infrastructure fully covers previously supported role-based authentication, plus it allows to configure any other possible authorization strategies including.

Reviewer: ```suggestion
The respective `AuthorizationManager` infrastructure fully covers the previously supported role-based authentication, plus it allows the configuration of any other possible authorization strategies.
```
